stan/math/prim/mat/fun/log1p.hpp

```diff
@@ -34,11 +35,25 @@ struct log1p_fun {
  * @param x Container.
  * @return Elementwise log1p of members of container.
  */
-template <typename T>
-inline typename apply_scalar_unary<log1p_fun, T>::return_t log1p(const T& x) {
+template <
+    typename T /*, typename = require_not_eigen_vt<std::is_arithmetic, T>*/>
+inline auto log1p(const T& x) {
   return apply_scalar_unary<log1p_fun, T>::apply(x);
 }
 
+// TODO(Tadej): This is inconsistent with stan's scalar version (which throws on
+// value <-1).
+///**
+// * Version of log1p() that accepts Eigen Matrix ar matrix expressions.
+// * @tparam Derived derived type of x
+// * @param x Matrix or matrix expression
+// * @return Elementwise log1p of members of container.
+// */
+// template <typename Derived, typename = require_eigen_vt<std::is_arithmetic,
+// Derived>>  inline auto log1p(const Eigen::MatrixBase<Derived>& x){
+//  return x.derived().array().log1p().matrix();
+//}
+
```

Comment:
I forgot to delete this. The problem is the scalar version of `log1p()` throws for invalid values. So we can not directely use eigen implementation that does not throw. Since eigen implementation is no faster that `applyScalarUnary` I decided not to complicate and leave this as is.